baseapp/abci.go

```diff
@@ -761,6 +761,10 @@ func (app *BaseApp) internalFinalizeBlock(ctx context.Context, req *abci.Request
 
 	events = append(events, beginBlock.Events...)
 
+	// Reset the gas meter so that the AnteHandlers aren't required to
+	gasMeter = app.getBlockGasMeter(app.finalizeBlockState.ctx)
+	app.finalizeBlockState.ctx = app.finalizeBlockState.ctx.WithBlockGasMeter(gasMeter)
+
```

Comment:
Personally, I think like that bot, so not sure if we can. However, it seems like it wouldn't change anything for those using the default ante handlers? So it is only a behavior change for those not using them? (Which still leaves the first question because of that use case)